GoogleMapsComponents/Maps/BaseListableEntity.cs

```diff
@@ -0,0 +1,106 @@
+﻿using Microsoft.JSInterop;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GoogleMapsComponents.Maps
+{
+    public abstract class BaseListableEntity<T> : IDisposable, IJsObjectRef
+        where T: BaseListableEntityOptions
+    {
+        protected readonly JsObjectRef _jsObjectRef;
+
+        public readonly Dictionary<string, List<MapEventListener>> EventListeners;
+
+        public Guid Guid => _jsObjectRef.Guid;
+
+        internal BaseListableEntity(JsObjectRef jsObjectRef)
+        {
+            _jsObjectRef = jsObjectRef;
+            EventListeners = new Dictionary<string, List<MapEventListener>>();
+        }
+
+        public void Dispose()
+        {
+            foreach (string key in EventListeners.Keys)
+            {
+                //Probably superfluous...
+                if (EventListeners[key] != null)
+                {
+                    foreach (MapEventListener eventListener in EventListeners[key])
+                    {
+                        eventListener.Dispose();
+                    }
+
+                    EventListeners[key].Clear();
+                }
+            }
+
+            EventListeners.Clear();
+            _jsObjectRef.Dispose();
+        }
+
+        public Task<Map> GetMap()
+        {
+            return _jsObjectRef.InvokeAsync<Map>(
+                "getMap");
+        }
+
+
+
+        /// <summary>
+        /// Renders the mao entity on the specified map or panorama. 
+        /// If map is set to null, the map entity will be removed.
+        /// </summary>
+        /// <param name="map"></param>
+        public async Task SetMap(Map map)
+        {
+            await _jsObjectRef.InvokeAsync(
+                   "setMap",
+                   map);
+
+            //_map = map;
+        }
+
+        public async Task<MapEventListener> AddListener(string eventName, Action handler)
+        {
+            JsObjectRef listenerRef = await _jsObjectRef.InvokeWithReturnedObjectRefAsync("addListener", eventName, handler);
+            MapEventListener eventListener = new MapEventListener(listenerRef);
+
+            if (!EventListeners.ContainsKey(eventName))
+            {
+                EventListeners.Add(eventName, new List<MapEventListener>());
+            }
+            EventListeners[eventName].Add(eventListener);
+
+            return eventListener;
+        }
+
+        public async Task<MapEventListener> AddListener<V>(string eventName, Action<V> handler)
```

Comment:
Personally i really really hate these short generic types constrains (i dont know how express it but you get point). Sometimes it is ok, but most times it is not. This time is V. What is v. If you looks at MS guys they use T or TSmth always
It is hart to touch them, replace, inspect since they are only one char.  Sometime you could even give meaning to it. Example TActionElement

Reply:
Yeah... V in this context is a generic T (originally was T, anything) but main class yet use T as generic, so compiler get a waring and since T and U are yet used, the following letter is "V".
I'll modify code with more clear type name, it is opportune.